enhancements/results-forwarding.md

```diff
@@ -0,0 +1,504 @@
+
+### Non-Goals
+
+* Deprecate the current CRD/PVC approach: There is still a use-case for this.
+
+* Expand the "evidence" that's currently gathered by the Compliance Operator. This is
+  needed but should be done as part of a separate initiative.
+
+* Implement an archiving solution. That's being discussed as a separate
+[enhancement](https://github.com/ComplianceAsCode/compliance-operator/pull/66).
+
+## Proposal
+
+The Compliance Operator will forward compliance results and remediation
+recommendations if the supplied forwarding endpoint is valid. The Compliance
+Operator will also continue to use CRDs and PVCs to store results,
+remediations, and evidence as it does today.
+
+A user may explicitly disable in-cluster storage if forwarding is enabled. The
+Compliance Operator will not support disabling PVC storage without a valid
+forwarding endpoint.
+
+This will require changing the **aggregator** to always forward via a gRPC API
+to a configured implementation. In this case, the **aggregator** will act as a
+gRPC client. The **resultserver** then becomes a gRPC server, subject to the
+evidence forwarding provider and will be renamed **evidence-persistor** and
+responsible for writing results.
+
+Forwarder implementations would be configured via the `ScanSetting` and they'd be called "providers".
+For backwards compatibility, we'd have defaults that would point to the current mode of operation
+of the Compliance Operator. e.g. a `pvc` provider for evidence storage.
+
+For more detail on the proposed API, see the *API Extensions* section.
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters, I'd like to run
+  compliance scans on my fleet in such a way that uses the minimal amount of
+  resources.
+
+* As a Site Reliability Engineer managing multiple clusters, I'd like to have
+  one place to view the compliance stance of my whole fleet.
+
+### API Extensions
+
+Given the scope of the change, we'd upgrade the `ScanSetting` version to
+`v1alpha2`. The following sections describe the additional values needed by the
+`ScanSetting` CRD.
+
+#### `resultForwarding.provider`
+
+Required: True
+Type: string
+
+The implementation to use for forwarding. The initial implementation will
+support `grpc`.
+
+#### `resultForwarding.grpc.endpoint`
+
+Required: True
+Type: string
+
+Endpoint URL to a gRPC server for result storage. The service may be in or
+outside the cluster. One potential implementation may be
+[compserv](https://github.com/rhmdnd/compserv).
+
+#### `resultForwarding.grpc.version`
+
+Required: False
+Default: `v1`
+Type: string
+
+#### `resultForwarding.grpc.tls.dynamic.caSecretName`
+
+Required: True
+Type: string
+
+Reference to a CA certificate and key pair for TLS.
+
+#### `resultForwarding.grpc.tls.dynamic.serverSecretName`
+
+Required: True
+Type: string
+
+Reference to a server certificate/keypair secret for mutual TLS.
```

Comment:
confused as per why do we need the server cert? I thought the aggregator was a client only?
Or is this secret meant for the GRPC service?